src/os/ui/geo/position.js

```diff
@@ -189,19 +189,23 @@ os.ui.geo.PositionCtrl.prototype.destroy_ = function() {
  * @private
  */
 os.ui.geo.PositionCtrl.prototype.onPosText_ = function() {
-  if (this.scope_['posText']) {
-    var result = os.geo.parseLatLon(this.scope_['posText'], this.scope_['order']);
+  const text = this.scope_['posText'];
+  if (text) {
+    var result = os.geo.parseLatLon(text, this.scope_['order']);
     if (result != null && Math.abs(result.lat) > 90) {
       // If the result isnt in range, set it to null to invalidate form
       result = null;
     }
     if (result == null) {
       try {
-        var coord = osasm.toLonLat(this.scope_['posText']);
-        result = /** @type {!osx.geo.Location} */ ({
-          lon: coord[0],
-          lat: coord[1]
-        });
+        const mgrs = text.replace(/\s+/g, '').toUpperCase();
```

Comment:
`os.ui.geo.geoLinkFn` will need similar logic, or this form needs to validate differently. Currently the position is parsed, but the field is still marked as invalid.

Reply:
Gotcha. I'll extract that out.